rewrite-java-test/src/test/java/org/openrewrite/java/format/WrappingAndBracesTest.java

```diff
@@ -508,4 +509,186 @@ public enum Status {
+              @interface Foo {
+                  @interface Foos {
+                      Foo[] value();
+                  }
+              }
+              """,
+            SourceSpec::skip),
+          java(
+            """
+              @Foo @Foo class Test {
+                  @Foo @Foo int field;
+              
+                  @Foo @Foo void method(
+                          @Foo
+                          @Foo
+                          int param) {
+                      @Foo
+                      @Foo
+                      int localVar;
+                  }
+              }
+              
+              enum MyEnum {
+                  @Foo
+                  @Foo
+                  VALUE
+              }
+              
+              record someRecord(
+                      @Foo
+                      @Foo
+                      String name) {
+              }
+              """,
+            """
+              @Foo
+              @Foo
+              class Test {
+                  @Foo
+                  @Foo
+                  int field;
+              
+                  @Foo
+                  @Foo
+                  void method(
+                          @Foo @Foo int param) {
+                      @Foo @Foo int localVar;
+                  }
+              }
+              
+              enum MyEnum {
+                  @Foo @Foo VALUE
+              }
+              
+              record someRecord(
+                      @Foo @Foo String name) {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void annotationWrappingModifiers() {
+        rewriteRun(
+          java(
+            """
+              import java.lang.annotation.Repeatable;
+              
+              @Repeatable(Foo.Foos.class)
+              @interface Foo {
+                  @interface Foos {
+                      Foo[] value();
+                  }
+              }
+              """,
+            SourceSpec::skip),
+          java(
+            """
+              @Foo @Foo final class Test {
+                  @Foo @Foo private int field;
+              
+                  @Foo @Foo public void method(
+                          @Foo
+                          @Foo
+                          final int param) {
+                      @Foo
+                      @Foo
+                      final int localVar;
+                  }
+              }
+              """,
+            """
+              @Foo
+              @Foo
+              final class Test {
+                  @Foo
+                  @Foo
+                  private int field;
+              
+                  @Foo
+                  @Foo
+                  public void method(
+                          @Foo @Foo final int param) {
+                      @Foo @Foo final int localVar;
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void annotationWrappingOther() {
```

Comment:
Rename this test. I don't know what "other" means here, so I can't neither check if the test makes sense.